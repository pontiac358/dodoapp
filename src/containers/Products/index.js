import React from 'react';
import PropTypes from 'prop-types';
import { connect } from "react-redux";
import { fetchProducts } from '../../actions/products'
import ProductList from './components/ProductList'


export class Products extends React.Component {

    componentDidMount () {
        this.handleFethProducts()
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.match.url !== this.props.match.url) {
            this.handleFethProducts(nextProps.match.params.id)
        }
    }

    handleFethProducts = (id) => {
        const categoryId = id ? id : this.props.match.params.id
        this.props.fetchProducts(categoryId)
    };


    render() {
        const { products } = this.props;

        return (
            <ProductList products={ products.data }/>
        );
    }
}

Products.propTypes = {
    products: PropTypes.shape({
        data: PropTypes.array,
        isLoading: PropTypes.bool
    }),
    cartProducts: PropTypes.array,
};

const mapStateToProps = (state, ownProps) => {
    return {
        products: state.products,
        ownProps
    }
};

const mapDispatchToProps = {
    fetchProducts
};

export default connect(mapStateToProps, mapDispatchToProps)(Products)