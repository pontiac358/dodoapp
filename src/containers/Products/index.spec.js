import React from 'react'
import { shallow } from 'enzyme'
import { Products } from '../Products'

describe('Products container', () => {
    const props = {
        products: {
            data: [],
            isLoading: false,
            errorMsg: null,
        },
        match: { params: { id:1 } },
        fetchProducts: () => {}
    };

    const wrapper = shallow(<Products { ...props } />);
    it("render correctly Products container", () => {
        expect(wrapper).toMatchSnapshot();
    });
});

