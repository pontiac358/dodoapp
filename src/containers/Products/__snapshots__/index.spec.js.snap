// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products container render correctly Products container 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Products
    fetchProducts={[Function]}
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
    products={
      Object {
        "data": Array [],
        "errorMsg": null,
        "isLoading": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "products": Array [],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "products": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
